web/modules/custom/dbcdk_community/src/Content/FlaggableContentRepository.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/**
+ * @file
+ * FlaggableContentRepository class definition.
+ */
+
+namespace Drupal\dbcdk_community\Content;
+
+use DBCDK\CommunityServices\Api\FlagApi;
+
+/**
+ * A repository for content
+ *
+ * @package Drupal\dbcdk_community\Content
+ */
+class FlaggableContentRepository {
+
+  /**
+   * The API to use when retrieving flagged content.
+   *
+   * @var FlagApi
+   */
+  protected $flagApi;
+
+  /**
+   * FlaggableContentRepository constructor.
+   *
+   * @param FlagApi $flag_api
+   *   The FlagApi class to use when retrieving content.
+   */
+  public function __construct(FlagApi $flag_api) {
+    $this->flagApi = $flag_api;
+  }
+
+  /**
+   * Get all content which has unread flags.
+   *
+   * The content will be sorted by reverse chronological order based on
+   * time of the latest flag ie. the content which has most recently been
+   * flagged will be returned first.
+   *
+   * @return FlaggableContent[]
+   *   All content with unread flags.
+   */
+  public function getContentWithUnreadFlags() {
+    /* @var FlaggableContent[] $content */
+    $content = [];
+
+    $filter = json_encode(['where' => ['markedAsRead' => FALSE]]);
+    $flags = $this->flagApi->flagFind($filter);
+
+    // Loop over *all* flags to find which content it is attached to.
+    // This is not very efficient but our generated APIs do not offer a
+    // better way to do this. In general the number of unread flags should
+    // not accumulate over time so it should be manageable.
```

Comment:
:+1: What about a "TODO" to remind us that this might be some sort of technical debt?